File: java/infrastructure/src/main/java/dev/ufadevconf/infrastructure/config/ApplicationConfig.java
package dev.ufadevconf.infrastructure.config;

import dev.ufadevconf.application.PickupPointApplicationService;
import dev.ufadevconf.domain.PickupPointRepository;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ApplicationConfig {

    @Bean
    public PickupPointApplicationService pickupPointApplicationService(PickupPointRepository repository) {
        return new PickupPointApplicationService(repository);
    }
}

File: kotlin/infrastructure/src/main/kotlin/dev/ufadevconf/infrastructure/config/ApplicationConfig.kt
package dev.ufadevconf.infrastructure.config

import dev.ufadevconf.application.PickupPointApplicationService
import dev.ufadevconf.domain.PickupPointRepository
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ApplicationConfig {

    @Bean
    fun pickupPointApplicationService(repository: PickupPointRepository): PickupPointApplicationService =
        PickupPointApplicationService(repository)
}
